harness/tests/experiment/pytorch/test_pytorch_trial.py

```diff
@@ -920,6 +927,501 @@ def mock_get_tensorboard_path(dummy: typing.Dict[str, typing.Any]) -> pathlib.Pa
 
         assert trial.legacy_counter.__dict__ == {"legacy_on_training_epochs_start_calls": 2}
 
+    @pytest.mark.skipif(torch.cuda.device_count() < 2, reason="not enough gpus")
+    @pytest.mark.gpu_parallel
+    def test_pytorch_parallel(self, tmp_path: pathlib.Path):
+        # set up distributed backend.
+        os.environ[det._DistributedBackend.TORCH] = str(1)
+
+        rdzv_backend = "c10d"
+        rdzv_endpoint = "localhost:29400"
+        rdzv_id = str(uuid.uuid4())
+
+        launch_config = launcher.LaunchConfig(
+            min_nodes=1,
+            max_nodes=1,
+            nproc_per_node=2,
+            run_id=rdzv_id,
+            max_restarts=0,
+            rdzv_endpoint=rdzv_endpoint,
+            rdzv_backend=rdzv_backend,
+        )
+
+        root_logfile = tmp_path.joinpath("root_test.log")
+
+        outputs = launcher.elastic_launch(launch_config, self.run_mnist)(tmp_path)
+        outputs = launcher.elastic_launch(launch_config, self.run_mnist)(tmp_path, outputs[0])
```

Comment:
You've got a _lot_ of copy/pasted code here, I think it's time for a helper function.

Reply:
created a new function in `pytorch_utils.py::setup_torch_distributed`